### DISCUSSION TITLE
Request for Integration of SingleFile with Zotero 7

### DISCUSSION THREAD

**Discussion by ghost** [gildas-lormeau/SingleFile#1554] [Ideas]
I hope this message finds you well.

I am writing to express my appreciation for SingleFile. It has been a fantastic tool for saving HTML pages. Recently, I have been contemplating how to better organize these saved HTML files. I would like to organize files saved with SingleFile using folders and tags. After careful consideration, I have concluded that utilizing Zotero, which is a database for organizing and storing references, would be an excellent solution. Zotero, released about a month ago, now includes a feature for previewing HTML files.

Given this, I kindly request if it would be possible to add functionality to SingleFile that allows for direct saving of files into Zotero 7. This would greatly enhance the organization and management of my saved HTML pages.

Thank you very much for your attention to this matter.

**Comment by gildas-lormeau** [comment]
Actually, Zotero uses SingleFile under the hood to produce page snaphots.

**Reply by kangruixiang** [reply]
Hi @gildas-lormeau, one of the things SingleFile can do is to save a selection. When saving a webpage in Zotero, the entire page gets saved, with sometimes only a small portion of the page being the actual content. Is there a way to do similar selection of element/div like you could in SingleFile?

**Comment by CoderJackZhu** [comment]
Seconded. Currently, when saving web pages with Zotero, only snapshots can be saved, which merely store the name of the web page without downloading the content to the local device. It would be much more convenient if this could be integrated. I hope this feature can be added.

**Reply by gildas-lormeau** [reply]
@CoderJackZhu As I said, Zotero already uses SingleFile under the hood to capture snapshots. Snapshots should be saved on the local device. If it does not work, then it's maybe a configuration issue or limitation/bug in Zotero.
